# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/LoadedScript.cpp
# Commit: a71a2a98b6bc
# Full Hash: a71a2a98b6bc017a7bb60d4b7c41f543f156eede
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move mElement from ScriptFetchOptions to DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D136742
# ==============================================================================

diff -r d52858ae58d2 -r a71a2a98b6bc dom/script/LoadedScript.cpp
--- a/dom/script/LoadedScript.cpp	Mon Feb 21 15:39:04 2022 +0000
+++ b/dom/script/LoadedScript.cpp	Mon Feb 21 15:39:05 2022 +0000
@@ -7,6 +7,7 @@
 #include "LoadedScript.h"
 
 #include "mozilla/HoldDropJSObjects.h"
+#include "mozilla/dom/Element.h"
 
 #include "jsfriendapi.h"
 #include "js/Modules.h"  // JS::{Get,Set}ModulePrivate
@@ -27,10 +28,11 @@
 NS_IMPL_CYCLE_COLLECTION_UNLINK_BEGIN(LoadedScript)
   NS_IMPL_CYCLE_COLLECTION_UNLINK(mFetchOptions)
   NS_IMPL_CYCLE_COLLECTION_UNLINK(mBaseURL)
+  NS_IMPL_CYCLE_COLLECTION_UNLINK(mElement)
 NS_IMPL_CYCLE_COLLECTION_UNLINK_END
 
 NS_IMPL_CYCLE_COLLECTION_TRAVERSE_BEGIN(LoadedScript)
-  NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mFetchOptions)
+  NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mFetchOptions, mElement)
 NS_IMPL_CYCLE_COLLECTION_TRAVERSE_END
 
 NS_IMPL_CYCLE_COLLECTION_TRACE_BEGIN(LoadedScript)
@@ -40,8 +42,11 @@
 NS_IMPL_CYCLE_COLLECTING_RELEASE(LoadedScript)
 
 LoadedScript::LoadedScript(ScriptKind aKind, ScriptFetchOptions* aFetchOptions,
-                           nsIURI* aBaseURL)
-    : mKind(aKind), mFetchOptions(aFetchOptions), mBaseURL(aBaseURL) {
+                           nsIURI* aBaseURL, Element* aElement)
+    : mKind(aKind),
+      mFetchOptions(aFetchOptions),
+      mBaseURL(aBaseURL),
+      mElement(aElement) {
   MOZ_ASSERT(mFetchOptions);
   MOZ_ASSERT(mBaseURL);
 }
@@ -91,16 +96,17 @@
 // EventScript
 //////////////////////////////////////////////////////////////
 
-EventScript::EventScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL)
-    : LoadedScript(ScriptKind::eEvent, aFetchOptions, aBaseURL) {}
+EventScript::EventScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
+                         Element* aElement)
+    : LoadedScript(ScriptKind::eEvent, aFetchOptions, aBaseURL, aElement) {}
 
 //////////////////////////////////////////////////////////////
 // ClassicScript
 //////////////////////////////////////////////////////////////
 
 ClassicScript::ClassicScript(ScriptFetchOptions* aFetchOptions,
-                             nsIURI* aBaseURL)
-    : LoadedScript(ScriptKind::eClassic, aFetchOptions, aBaseURL) {}
+                             nsIURI* aBaseURL, Element* aElement)
+    : LoadedScript(ScriptKind::eClassic, aFetchOptions, aBaseURL, aElement) {}
 
 //////////////////////////////////////////////////////////////
 // ModuleScript
@@ -129,8 +135,9 @@
 NS_IMPL_ADDREF_INHERITED(ModuleScript, LoadedScript)
 NS_IMPL_RELEASE_INHERITED(ModuleScript, LoadedScript)
 
-ModuleScript::ModuleScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL)
-    : LoadedScript(ScriptKind::eModule, aFetchOptions, aBaseURL),
+ModuleScript::ModuleScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
+                           Element* aElement)
+    : LoadedScript(ScriptKind::eModule, aFetchOptions, aBaseURL, aElement),
       mDebuggerDataInitialized(false) {
   MOZ_ASSERT(!ModuleRecord());
   MOZ_ASSERT(!HasParseError());